### PR TITLE
More general sample process

### PR DESCRIPTION
**Pull request by tilkb**
Could't predict from 1 element, if the batch is more than 1. It's fixed.

### CODE REVIEW

**Comment by fchollet** [comment]
This may work for Theano but it definitely wouldn't for TF. Please find a way that works for both backends.

File: examples/variational_autoencoder.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 def sampling(args):
     z_mean, z_log_var = args
-    epsilon = K.random_normal(shape=(batch_size, latent_dim), mean=0.,
+    epsilon = K.random_normal(shape=(K.shape(z_mean)[0], K.shape(z_mean)[1]), mean=0.,
```
Comment (joelthchao) [inline examples/variational_autoencoder.py]: This works in both tf and th, please help to check
```python 
epsilon = K.random_normal(shape=(K.int_shape(z_mean)[0], K.int_shape(z_mean)[1]), mean=0.,
```

Comment (javiercorrea) [inline examples/variational_autoencoder.py]: How about:

    epsilon = K.random_normal(shape=K.shape(z_mean), mean=0.,

this works on Tensorflow CPU.

**Comment by fchollet** [comment]
Closing inactive PR.

**Pull request closed without merging**